libmscore/segment.cpp

```diff
@@ -1062,6 +1062,20 @@ bool Segment::hasElements(int minTrack, int maxTrack) const
       return false;
       }
 
+//---------------------------------------------------------
+//   allElementsInvisible
+///  return true if all elements in the segment are invisible
+//---------------------------------------------------------
+
+bool Segment::allElementsInvisible() const
```

Comment:
Would there be a way to only do this once during layout, like at the same time we are marking the segment itself visible?  I guess maybe this will turn out to be only once, but still, it's an extra loop through the elements.